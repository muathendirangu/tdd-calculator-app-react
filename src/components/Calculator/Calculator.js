import React, { Component } from 'react';
import './Calculator.css';

class Calculator extends Component {
    state = {
        // value to be displayed in the display component <Display />
        displayValue : '0',
        // Values to be displayed as number in the key component <Key />
        numbers : [],
        // values to be displayed as the operator in the key component <Key />
        operators : [],
        // variable to store the selected  math operation
        selectedOperator: '',
        // stored value to be used for math operation
        storedValue: '',
    }

    callOperator = () => {
        console.log('execute math operation');        
    }

    setOperator = () => {
        console.log('choose appropriate operator');
    }
    
    updateDisplay = () => {
        console.log('modify display');
    }

    render = () => {
        return (
            <div className="calculator-container"/>
        );
    }
}

export default Calculator;