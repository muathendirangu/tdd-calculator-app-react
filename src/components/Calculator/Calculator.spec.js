import React from 'react';
import { shallow } from 'enzyme';
import Calculator from './Calculator';

describe('Calculator', () => {
    let wrapper;
    beforeEach(() => wrapper = shallow(<Calculator />));

    it('takes a snapshot of Calculator component', () => expect(wrapper).toMatchSnapshot());
    it('should render a <div />', () => {
      expect(wrapper.find('div').length).toEqual(1);
    });

    it('should render the display component', () => {
      expect(wrapper.containsMatchingElement(<Display/>)).toEqual(true);
    })
  });
  