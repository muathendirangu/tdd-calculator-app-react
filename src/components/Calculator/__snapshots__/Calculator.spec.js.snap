// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator takes a snapshot of Calculator component 1`] = `ShallowWrapper {}`;
